app/models/user.rb

```diff
@@ -51,7 +51,7 @@ def memberships_for(action, klass)
   def self.scope_for(action, user, opts={})
     case action
     when :show, :index
-      all
+      where(activated_state: 0)
```

Comment:
I think I did it one better those scope declarations were already provided
by the enum declaration.
On Jan 30, 2015 3:59 PM, "Campbell Allen" <notifications@github.com> wrote:

> In app/models/user.rb
> <https://github.com/zooniverse/Panoptes/pull/470#discussion_r23875794>:
>
> > @@ -51,7 +51,7 @@ def memberships_for(action, klass)
> >    def self.scope_for(action, user, opts={})
> >      case action
> >      when :show, :index
> > -      all
> > +      where(activated_state: 0)
>
> Can you try and use the activatable scopes here:
> https://github.com/zooniverse/Panoptes/blob/master/app/models/concerns/activatable.rb#L6
>
> They need to change to:
>
> scope :active, -> { where(activated_state: activated_states[:active]) }
> scope :disabled, -> { where(activated_state: activated_states[:inactive]) }
>
> Does this still get used in the controller actions?
> https://github.com/zooniverse/Panoptes/blob/master/app/controllers/api/v1/users_controller.rb#L34
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zooniverse/Panoptes/pull/470/files#r23875794>.
>